elements/bulbs-video/bulbs-video.js

```diff
@@ -16,23 +14,57 @@ import 'videojs/dist/video-js/video-js.css';
 import 'videohub-player/dist/videohub-player.css';
 import 'videojs-autoplay-toggle/videojs.autoplay-toggle.css';
 
+// let videoStores = {};
+
 export default class BulbsVideo extends BulbsElement {
   initialDispatch () {
     this.store.actions.fetchVideo(this.props.src);
+    if (typeof this.props.autoplay === 'string') {
+      this.store.actions.revealPlayer();
+    }
   }
 
-  componentWillReceiveProps (props) {
-    if (this.props.src !== props.src) {
+  componentDidUpdate (prevProps) {
+    if (this.props.src !== prevProps.src) {
       this.store.actions.resetController();
-      this.store.actions.fetchVideo(props.src);
+      this.initialDispatch();
+    }
+  }
+
+/*
```

Comment:
Next task for me related to the video player involves getting redundant requests for the same resource trimmed down. Going to send out an RFC before diving into that in earnest.